realm/realm-library/src/main/cpp/io_realm_internal_OsObjectSchemaInfo.cpp

```diff
@@ -109,3 +109,18 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeGetPrope
     CATCH_STD()
     return reinterpret_cast<jlong>(nullptr);
 }
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeGetPrimaryKeyProperty(JNIEnv*, jclass,
+                                                                                              jlong native_ptr)
+{
+    TR_ENTER_PTR(native_ptr)
+
+    // no throws
+    auto& object_schema = *reinterpret_cast<ObjectSchema*>(native_ptr);
+    auto* property = object_schema.primary_key_property();
+    if (property) {
+        return reinterpret_cast<jlong>(new Property(*property));
```

Comment:
`new` can throw, right?

Reply:
OK ... it throws `bad_alloc` ... will change this.